cmd/snap-update-ns/change.go

```diff
@@ -498,7 +498,13 @@ func neededChangesImpl(currentProfile, desiredProfile *osutil.MountProfile) []*C
 		if reuse[current[i].Dir] {
 			changes = append(changes, &Change{Action: Keep, Entry: current[i]})
 		} else {
-			changes = append(changes, &Change{Action: Unmount, Entry: current[i]})
+			var entry osutil.MountEntry = current[i]
+			entry.Options = append([]string(nil), entry.Options...)
+			// Detach directory bind mounts and filesystem mounts.
```

Comment:
IIRC these could be files too, also tmpfs is the only fs we mount. The comment should probably say `// Detach bind mounts and tmpfs mounts`

Reply:
Yes although files don't have this problem. We technically only mount `tmpfs` but the property holds for all filesystems, since they can by definition hold further mount entries underneath. I will expand the comment to explain why this is done.